Mac build issue: unknown type name `uuid_string_t`

Seen in #8255 (and #8233: https://discourse.julialang.org/t/binarybuild-failure-on-mac/111216):

```
[02:08:26] In file included from /opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/HFSVolumes.h:25:
[02:08:26] /opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/include/hfs/hfs_format.h:794:2: error: unknown type name 'uuid_string_t'; did you mean 'io_string_t'?
[02:08:26]         uuid_string_t   ext_jnl_uuid;
[02:08:26]         ^
```

This appears when building with the previous Pango recipe too, so it seems like a BinaryBuilder/Yggdrasil issue?

--- a/L/Libuuid/build_tarballs.jl
+++ b/L/Libuuid/build_tarballs.jl
@@ -24,6 +24,9 @@ make install
 # platforms are passed in on the command line
 platforms = supported_platforms()
 filter!(!Sys.iswindows, platforms)
+# This package on macOS creates more problems than it solves:
+# <https://github.com/JuliaPackaging/Yggdrasil/issues/8256>.
+filter!(!Sys.isapple, platforms)
 
 # The products that we will ensure are always built
 products = [

